src/app/Dashboard/LayoutTemplatePicker.tsx

```diff
@@ -0,0 +1,578 @@
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import { DeleteWarningModal } from '@app/Modal/DeleteWarningModal';
+import { DeleteOrDisableWarningType } from '@app/Modal/DeleteWarningUtils';
+import { FeatureFlag } from '@app/Shared/FeatureFlag/FeatureFlag';
+import { RootState, dashboardConfigDeleteTemplateIntent } from '@app/Shared/Redux/ReduxStore';
+import { ServiceContext } from '@app/Shared/Services/Services';
+import { fakeChartContext, fakeServices } from '@app/utils/fakeData';
+import { portalRoot } from '@app/utils/utils';
+import {
+  Button,
+  Card,
+  DescriptionList,
+  DescriptionListDescription,
+  DescriptionListGroup,
+  DescriptionListTerm,
+  Divider,
+  Drawer,
+  DrawerActions,
+  DrawerCloseButton,
+  DrawerContent,
+  DrawerContentBody,
+  DrawerHead,
+  DrawerPanelContent,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateIcon,
+  EmptyStateVariant,
+  Grid,
+  GridItem,
+  Select,
+  SelectOption,
+  SelectOptionObject,
+  SelectVariant,
+  Stack,
+  StackItem,
+  Title,
+  Toolbar,
+  ToolbarContent,
+  ToolbarFilter,
+  ToolbarGroup,
+  ToolbarItem,
+  ToolbarToggleGroup,
+} from '@patternfly/react-core';
+import {
+  ArrowsAltVIcon,
+  FilterIcon,
+  GlobeIcon,
+  InfoCircleIcon,
+  LongArrowAltDownIcon,
+  LongArrowAltUpIcon,
+  PficonTemplateIcon,
+  UploadIcon,
+} from '@patternfly/react-icons';
+import { InnerScrollContainer, OuterScrollContainer } from '@patternfly/react-table';
+import React from 'react';
+
+import { useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+import { ChartContext } from './Charts/ChartContext';
+import { JFRMetricsChartCardDescriptor } from './Charts/jfr/JFRMetricsChartCard';
+import { CryostatLayoutTemplates, BlankLayout } from './cryostat-dashboard-templates';
+import { getConfigByName, hasConfigByName } from './Dashboard';
+import { LayoutTemplate, LayoutTemplateContext, LayoutTemplateRecord, recordToLayoutTemplate } from './dashboard-utils';
+import { LayoutTemplateGroup } from './LayoutTemplateGroup';
+import { SearchAutocomplete } from './SearchAutocomplete';
+
+export type LayoutTemplateFilter = 'Suggested' | 'Cryostat' | 'User-submitted';
+
+export type LayoutTemplateSort = 'Name' | 'Card Count'; // TODO: add 'Version' after more version are released
+
+const TemplateSortSelectOption: React.FC<{ sort: LayoutTemplateSort }> = ({ sort }) => {
```

Comment:
Maybe we don't need this if it just pass straight down SelectOption without doing any other things?

Reply:
To be descriptive, we can convert the string union to a string enum?